src/components/QuickDropIcon.tsx

```diff
@@ -11,31 +13,31 @@ import { AlertText, AlertType } from '../constants'
 import theme from '../selectors/theme'
 import store from '../stores/app'
 
+export interface QuickDropIconProps {
+  alertType: AlertType
+  Icon: FC<IconType>
+  onHoverMessage: string | ((state: State, zone: DragThoughtZone) => string)
+  drop: (monitor: DropTargetMonitor) => void
+}
+
 /** Creates the props for drop. */
-const dropCollect = (connect: DropTargetConnector, monitor: DropTargetMonitor) => ({
-  dropTarget: connect.dropTarget(),
+const dropCollect = (monitor: DropTargetMonitor) => ({
   isDragInProgress: !!monitor.getItem(),
   zone: monitor.getItem()?.zone || null,
   isHovering: monitor.isOver({ shallow: true }),
 })
 
 /** The inner quick drop component that has been wrapped in a DropTarget. */
-const DroppableQuickDropIcon = ({
-  alertType,
-  dropTarget,
-  Icon,
-  isDragInProgress,
-  isHovering,
-  onHoverMessage,
-  zone,
-}: {
-  alertType: AlertType
-  Icon: FC<IconType>
-  onHoverMessage: string | ((state: State, zone: DragThoughtZone) => string)
-} & ReturnType<typeof dropCollect>) => {
+const DroppableQuickDropIcon = ({ alertType, Icon, onHoverMessage, drop }: QuickDropIconProps) => {
```

Comment:
You can combine `QuickDropIcon` and `DroppableQuickDropIcon` into one component. The separation no longer makes sense.